src/validators/problem.ts

```diff
@@ -14,21 +14,23 @@ export function problemValidator(method: string): ValidationChain[] {
         case "POST /problems": {
             return [
                 body("title", "Invalid or missing 'title'").isString().exists(),
-                body("source", "Invalid or missing 'source'").isString().exists(),
+                body("source", "Invalid or missing 'source'").isString().exists().custom(validateProblemPlatform),
                 body("sourceLink", "Invalid or missing 'sourceLink'").isString().exists(),
                 body("problemSetIds", "Invalid or missing 'problemSetId'").isArray().exists(),
                 body("problemSetIds", "Invalid or nonexistent 'problemSetId'").custom(validateProblemSetIds),
-                body("problemMetadata", "Invalid or missing 'problemMetadata' (optional)").optional()
+                body("problemMetadata", "Missing 'problemMetadata'").exists(),
+                body("problemMetadata", "Invalid or missing fields in 'problemMetadata'").custom(validateProblemMetadata)
             ];
         }
         case "PUT /problems/:problemId": {
             return [
                 param("problemId", "Invalid or missing ':problemId'").exists().isMongoId(),
                 body("title", "Invalid or missing 'title'").isString().exists(),
-                body("source", "Invalid or missing 'source'").isString().exists(),
+                body("source", "Invalid or missing 'source'").isString().exists().custom(validateProblemPlatform),
                 body("sourceLink", "Invalid or missing 'sourceLink'").isString().exists(),
                 body("problemSetIds", "Invalid or missing 'problemSetId'").isArray().exists(),
-                body("problemMetadata", "Invalid or missing 'problemMetadata' (optional)").optional()
+                body("problemSetIds", "Invalid or nonexistent 'problemSetId'").custom(validateProblemSetIds),
+                body("problemMetadata", "Invalid or missing fields in 'problemMetadata'").custom(validateProblemMetadata)
```

Comment:
should we add `.exists()` checks for these custom ones too?

Reply:
For `problemSetIds`, why have two different lines of validations returning the same message? The custom one should return a more specific message if we split them like that.

Reply:
They are different messages 'Invalid or Missing' vs 'Invalid or nonexistent'